google_maps/public/js/src/google_maps_edit.js

```diff
@@ -47,10 +71,23 @@ function GoogleMapsEditXBlock(runtime, element) {
     clearButton.prop('hidden', true);
   }
 
-  $(document).ready(function() {
+  /**
+   * Checks if the value was selected from Google Maps autocomplete dropdown
+   */
+  function validatePlaceNameField() {
+    if (iframePlaceNameInput.val() === googleMapsXBlockApp.place.name ) {
+      googleMapsXBlockApp.toggleErrorMessage($('#place-name-input'), false);
+      saveXBlockData();
+    } else {
+      googleMapsXBlockApp.toggleErrorMessage($('#place-name-input'), true);
+    }
+  }
+
+  $(function() {
     displayNameClearButton.prop('hidden', displayNameInput.val() === displayNameDefaultValue);
     iframeWidthClearButton.prop('hidden', iframeWidthInput.val() === iframeWidthDefaultValue);
     iframeHeightClearButton.prop('hidden', iframeHeightInput.val() === iframeHeightDefaultValue);
+    iframePlaceNameClearButton.prop('hidden', iframePlaceNameInput.val() === iframePlaceNameDefaultValue);
   });
```

Comment:
On load, handle the errors returned by the Google Maps API. For example: if the API key is invalid, API is not enabled.
Simply show messages like:
`Invalid Google Maps API key. Please check your settings.`
or
`{API name} is not enabled. Please check the Google Developers Console.`